nntrainer/tensor/memory_pool.cpp

```diff
@@ -250,6 +250,22 @@ size_t MemoryPool::calcMinMemoryRequirement() {
                         return val1.second < val2.second;
                       });
   unsigned int last_interval = max_interval.second;
+  /** as weights stay valid for max duration, ignore this value and get the real
+   * max value */
+  if (last_interval == std::numeric_limits<unsigned int>::max()) {
+    max_interval = *std::max_element(
+      memory_validity.begin(), memory_validity.end(),
+      [last_interval](auto const &val1, auto const &val2) {
+        return ((val2.second != last_interval) && (val1.second < val2.second));
```

Comment:
Is this calculation assumes that the given vector is sorted in some order?

Reply:
no, this doesn't assume any sorting order. This is trying to get the max value of the validity to determine the range of the validity. So, if the max value is UINT_MAX, it gets the 2nd highest value.

Reply:
[7, 8]
[4, 6]
[4, MAX]

In this case, would [7, 8] be selected?

Reply:
yes.

Reply:
note that as we are only interested in the end_value, so effectively 8 is selected from the interval.

Reply:
I see, thanks :)